test/lib/ansible_test/_internal/payload.py

```diff
@@ -91,6 +91,8 @@ def detect_permissions(tar_info: tarfile.TarInfo) -> t.Optional[tarfile.TarInfo]
         """
         if tar_info.path.startswith('ansible/'):
             mode = permissions.get(os.path.relpath(tar_info.path, 'ansible'))
+        elif data_context().content.collection and tar_info.path.startswith(data_context().content.collection.directory):
```

Comment:
This should probably be
```suggestion
        elif data_context().content.collection and tar_info.path.startswith(data_context().content.collection.directory + '/'):
```